WooCommerce/WooCommerceTests/POS/Models/PointOfSaleAggregateModelTests.swift

```diff
@@ -154,144 +154,66 @@ struct PointOfSaleAggregateModelTests {
         private let sut: PointOfSaleAggregateModel
 
         init() {
-            orderService.orderToReturn = Order.fake()
-
+            orderController.orderStateToReturn = makeLoadedOrderState(cartTotal: "$0.00")
             sut = PointOfSaleAggregateModel(
                 itemsService: itemsService,
                 cardPresentPaymentService: cardPresentPaymentService,
-                orderService: orderService)
+                orderController: orderController)
 
             sut.addToCart(makeItem())
         }
 
-        @Test func startNewCart_sets_orderState_to_idle() async throws {
+        @Test func startNewCart_calls_clearOrder() async throws {
             // Given
-            await sut.checkOut()
-            try #require(sut.orderState == .loaded(.init(
-                cartTotal: "$0.00",
-                orderTotal: "",
-                taxTotal: "")))
 
             // When
             sut.startNewCart()
 
             // Then
-            #expect(sut.orderState == .idle)
-        }
-
-        @Test func checkOut_when_reader_connects_collectPayment_called() async throws {
-            // Given
-            cardPresentPaymentService.connectedReader = nil
-            await sut.checkOut()
-            cardPresentPaymentService.collectPaymentWasCalled = false
-
-            // When
-            // `await confirmation` callback only waits until this completes, not until some timeout.
-            // Since this is synchonous but triggers async combine behaviour, we can't use that approach.
-            cardPresentPaymentService.connectedReader = .init(name: "Test reader", batteryLevel: 0.7)
-
-            // Then
-            let timeout = ContinuousClock.now + .seconds(2)
-
-            while cardPresentPaymentService.collectPaymentWasCalled != true {
-                try await Task.sleep(for: .milliseconds(1))
-                try #require(.now < timeout)
-            }
+            #expect(orderController.clearOrderWasCalled == true)
         }
 
-        @Test func checkOut_when_reader_is_already_connected_collectPayment_called() async throws {
+        @Test func checkout_with_items_calls_sync_order() async throws {
             // Given
-            cardPresentPaymentService.connectedReader = .init(name: "Test reader", batteryLevel: 0.7)
-            orderService.orderToReturn = Order.fake().copy(items: [.fake()])
+            sut.addToCart(makeItem())
+            let item = try #require(sut.cart.first)
 
             // When
             await sut.checkOut()
 
-            // Then
-            #expect(cardPresentPaymentService.collectPaymentWasCalled)
+            let passedItem = try #require(orderController.spyCartProducts?.first)
+            #expect(passedItem.id == item.id)
         }
 
-        @Test func after_disconnection_when_reader_reconnects_collectPayment_called() async throws {
+//        The UI prevents no-item checkouts, but it's the controller's responsibility to handle this.
```

Comment:
super nit: maybe just align the indentation with the function so that it doesn't feel like a commented out line?

```suggestion
        // The UI prevents no-item checkouts, but it's the controller's responsibility to handle this.
```